### PR TITLE
Less restrictive rules for Foundry cheatcodes.

### PR DESCRIPTION
**Pull request by david-k**
Fixes https://github.com/runtimeverification/foundry-tests/issues/23

Previously, cheatcodes where matched against the following CALL instruction:

  CALL _ CHEAT_ADDR 0 ARGSTART _ARGWIDTH 0 0

In this commit, this is changed as follows:

  CALL _ CHEAT_ADDR _VALUE ARGSTART _ARGWIDTH _RETSTART _RETWIDTH

In particular:
1. The memory region that contains the return value may now be arbitrary (_RETSTART and _RETWIDTH)
2. An arbitrary amount of ether may be sent when calling the cheatcode (_VALUE)

Because RETSTART and RETWIDTH were initially required to be zero, the K rules where actually never applied. This is because the bytecode emitted by the Solidity compiler always sets RETSTART to a non-zero value, even for functions that do not return any value.

Allowing arbitrary ether to be sent along with cheatcodes ensures we are consistent with Foundry. Note that the ether is not actually transferred but simply ignored. This, too, is consistent with Foundry.

### CODE REVIEW

**Review by lisandrasilva — APPROVED** [review]

**Review by JuanCoRo — APPROVED** [review]
Looks good! The only thing I would change is to not include the commit from #1356 and instead merge `master` into the PR's branch to encapsulate the changes more clearly.

**Comment by david-k** [comment]
Yeah, sorry, was playing around with git :-)

File: foundry.md [hunk 1]
```diff
@@ -269,7 +269,7 @@ This rule then takes the `uint256` value using `#asWord(#range(LM, ARGSTART +Int
 
 ```k
     rule [call.coinbase]:
-         <k> CALL _ CHEAT_ADDR 0 ARGSTART _ARGWIDTH 0 0 => 1 ~> #push ... </k>
+         <k> CALL _ CHEAT_ADDR _VALUE ARGSTART _ARGWIDTH _RETSTART _RETWIDTH => 1 ~> #push ... </k>
          <output> _ => .ByteArray </output>
          <localMem> LM </localMem>
          <coinbase> _ => #asWord(#range(LM, ARGSTART +Int 4, 32)) </coinbase>
```
Comment (ehildenb) [inline foundry.md:275]: Did we check the foundry source code for matching here? For eaxmple, does the foundry source code allow any value? or only zero value?

**Comment by ehildenb** [comment]
Did we check that these cheat codes are actually being applied? This can be done in the `kore-repl`, and we should check at least a handful of them.

**Comment by ehildenb** [comment]
That's a fine way to do it, exactyl how I would approach it, but I would have just turned on the logging for `kore-exec`, and avoided stepping into the repl. Something like `--log-entries DebugAttempRule --log log.file` added as arguments to `kore-exec`, then search thorugh teh log file produced.

**Comment by david-k** [comment]
Just tried this with `kevm`, and it seems you can use `--haskell-backend-arg` to pass arguments to `kore-exec`. Nice!

**Pull request merged**